pkg/oc/admin/diagnostics/diagnostics.go

```diff
@@ -126,19 +127,50 @@ func NewCmdDiagnostics(name string, fullName string, out io.Writer) *cobra.Comma
 	o.bindClientFlags(cmd.Flags())
 	o.bindHostFlags(cmd.Flags())
 
-	// add "all" subcommand
-	cmd.AddCommand(NewCmdDiagnosticsAll(AllDiagnosticsRecommendedName, fullName+" "+AllDiagnosticsRecommendedName, out, available))
-	// add individual diagnostic subcommands
-	for _, diag := range available {
-		cmd.AddCommand(NewCmdDiagnosticsIndividual(strings.ToLower(diag.Name()), fullName+" "+strings.ToLower(diag.Name()), out, diag))
+	groups := createCommnadGroups(fullName, out)
+	groups = append(groups, ktemplates.CommandGroup{"All Diagnostics:",
+		[]*cobra.Command{
+			NewCmdDiagnosticsAll(AllDiagnosticsRecommendedName, fullName+" "+AllDiagnosticsRecommendedName, out, available),
+		},
+	})
+	groups.Add(cmd)
+	templates.ActsAsRootCommand(cmd, []string{"options"}, groups...)
+
+	return cmd
+}
+
+// createCommandGroups creates a group of sub commands to make help option clear.
+func createCommnadGroups(fullName string, out io.Writer) ktemplates.CommandGroups {
+	groups := ktemplates.CommandGroups{}
+
+	// Client Diagnostics
+	cmd := []*cobra.Command{}
+	for _, diag := range availableClientDiagnostics() {
+		cmd = append(cmd, NewCmdDiagnosticsIndividual(strings.ToLower(diag.Name()), fullName+" "+strings.ToLower(diag.Name()), out, diag))
+	}
+	groups = append(groups, ktemplates.CommandGroup{"Clinent Diagnostics:", cmd})
+
+	// Cluster Diagnostics
+	cmd = []*cobra.Command{}
+	for _, diag := range availableClusterDiagnostics() {
+		cmd = append(cmd, NewCmdDiagnosticsIndividual(strings.ToLower(diag.Name()), fullName+" "+strings.ToLower(diag.Name()), out, diag))
 	}
-	// add hidden in-pod subcommands
-	cmd.AddCommand(
+	groups = append(groups, ktemplates.CommandGroup{"Cluster Diagnostics:", cmd})
+
+	// Host Diagnostics
+	cmd = []*cobra.Command{}
+	for _, diag := range availableHostDiagnostics() {
+		cmd = append(cmd, NewCmdDiagnosticsIndividual(strings.ToLower(diag.Name()), fullName+" "+strings.ToLower(diag.Name()), out, diag))
+	}
+	groups = append(groups, ktemplates.CommandGroup{"Host Diagnostics:", cmd})
+
+	// Network Diagnostics
+	cmd = []*cobra.Command{
 		poddiag.NewCommandPodDiagnostics(poddiag.InPodDiagnosticRecommendedName, out),
 		networkpoddiag.NewCommandNetworkPodDiagnostics(networkpoddiag.InPodNetworkCheckRecommendedName, out),
-	)
-
-	return cmd
+	}
+	groups = append(groups, ktemplates.CommandGroup{"Netowrk Diagnostics:", cmd})
```

Comment:
typo Netowrk -> Network
however these are supposed to be hidden from the user, never invoked directly. Don't expose them.